diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc2dcbe..3133dae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,15 +35,18 @@ list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 # options
 option(QCM_BUILD_TESTS "Build tests" ${PROJECT_IS_TOP_LEVEL})
 option(QCM_USE_ASAN "use asan" OFF)
+option(QCM_USE_LTO "use lto" ON)
 option(QCM_QML_DEBUG "qml debug" OFF)
 
-# check lto
-include(CheckIPOSupported)
-check_ipo_supported(RESULT is_ipo_supported OUTPUT lto_error)
-if(is_ipo_supported)
-  message(STATUS "enable lto")
-  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
-  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_DEBUG OFF)
+if (QCM_USE_LTO)
+  # check lto
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT is_ipo_supported OUTPUT lto_error)
+  if(is_ipo_supported)
+    message(STATUS "enable lto")
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_DEBUG OFF)
+  endif()
 endif()
 
 if(UNIX AND NOT APPLE)
