diff --git a/src/sys/pal/unix/futex.cpp b/src/sys/pal/unix/futex.cpp
index 441bb9a..1bef28f 100644
--- a/third_party/rstd/src/sys/pal/unix/futex.cpp
+++ b/third_party/rstd/src/sys/pal/unix/futex.cpp
@@ -1,6 +1,6 @@
-#ifdef __unix__
 module;
 
+#ifdef __unix__
 #    include <cerrno>
 #    include <chrono>
 #    include <cstdio>
@@ -14,9 +14,11 @@ module;
 #    elif defined(__FreeBSD__)
 #        include <sys/umtx.h>
 #    endif
+#endif
 module rstd.sys;
 import :pal.unix.futex;
 
+#ifdef __unix__
 namespace rstd::pal::unix
 {
 
diff --git a/src/sys/pal/unix/futex.cppm b/src/sys/pal/unix/futex.cppm
index 91b19d6..b3ebd33 100644
--- a/third_party/rstd/src/sys/pal/unix/futex.cppm
+++ b/third_party/rstd/src/sys/pal/unix/futex.cppm
@@ -1,7 +1,8 @@
+module;
+
 #ifndef __unix__
 export module rstd.sys:pal.unix.futex;
 #else
-module;
 #    include <atomic>
 #    include <chrono>
 #    include <cstdint>
diff --git a/src/sys/pal/windows/futex.cpp b/src/sys/pal/windows/futex.cpp
index 3853b4a..cf18b56 100644
--- a/third_party/rstd/src/sys/pal/windows/futex.cpp
+++ b/third_party/rstd/src/sys/pal/windows/futex.cpp
@@ -1,10 +1,13 @@
-#ifdef _WIN32
 module;
+
+#ifdef _WIN32
 #    include <windows.h>
 #    include <atomic>
+#endif
 module rstd.sys;
 import :pal.windows.futex;
 
+#ifdef _WIN32
 namespace rstd::pal::windows
 {
 bool wait_on_address(const void* address, const void* compare, size_t size, DWORD timeout) {
diff --git a/src/sys/pal/windows/futex.cppm b/src/sys/pal/windows/futex.cppm
index 90e3088..394bb62 100644
--- a/third_party/rstd/src/sys/pal/windows/futex.cppm
+++ b/third_party/rstd/src/sys/pal/windows/futex.cppm
@@ -1,7 +1,8 @@
+module;
+
 #ifndef _WIN32
 export module rstd.sys:pal.windows.futex;
 #else
-module;
 #    include <atomic>
 #    include <chrono>
 #    include <cstdint>
