--- a/setup.py	2024-07-13 19:24:16.001303928 +0800
+++ b/setup.py	2024-07-13 19:24:40.679695152 +0800
@@ -1,6 +1,6 @@
 import os
 
-from setuptools import find_packages, setup
+from setuptools import find_namespace_packages, setup
 
 from mcdreforged.constants import core_constant
 
@@ -65,7 +65,7 @@
 	author=AUTHOR,
 	python_requires=REQUIRES_PYTHON,
 	project_urls=PROJECT_URLS,
-	packages=find_packages(exclude=['tests', '*.tests', '*.tests.*', 'tests.*']),
+	packages=find_namespace_packages(exclude=['tests', '*.tests', '*.tests.*', 'tests.*']),
 	include_package_data=True,
 	install_requires=REQUIRED,
 	classifiers=CLASSIFIERS,
