--- a/mcdreforged/mcdr_server.py	2024-07-13 18:08:11.731580563 +0800
+++ b/mcdreforged/mcdr_server.py	2024-07-13 18:10:26.344782844 +0800
@@ -83,7 +83,8 @@
 		self.preference_manager: PreferenceManager = PreferenceManager(self)
 		self.__tr = self.create_internal_translator('mcdr_server')
 
-		self.__check_environment()
+		# Skip this to fix environment detection
+		# self.__check_environment()
 
 		# --- Input arguments "generate_default_only" processing --- #
 		if args.generate_default_only:
